### ISSUE TITLE
Unable to build project with numpy

### ISSUE THREAD

**Issue by capcarr** [rtfd/readthedocs.org#1624]
I'm unable to build the documentation for my project, which depends on numpy (and scipy and matplotlib). The build fails with a RequirementParseError for numpy. Here's the log: https://readthedocs.org/projects/biosppy/builds/3295858/

Under Tools->Advanced Settings I have both the options "Install your project inside a virtualenv using setup.py install" and "Give the virtual environment access to the global site-packages dir." checked.

**Comment by agjohnson** [comment]
I can replicate the issue locally with `pip==7.1.0` and your requirements file. The error is from pip, which is stating you have a malformed spec for `numpy`. I'm not sure what makes the spec malformed here though. 

Closing as this isn't a Read the Docs issue.

**Comment by capcarr** [comment]
OK, thanks.

Can you suggest any workarounds in the mean time?

**Comment by agjohnson** [comment]
Try a standard `pip freeze` in your virtualenv to create a valid `requirements.txt`

**Comment by capcarr** [comment]
For future reference, I was only able to build my documentation by specifying an empty requirements file in RTD Admin, and by mocking any 3rd party packages in `conf.py`, as is described in the [FAQ](https://read-the-docs.readthedocs.org/en/latest/faq.html#i-get-import-errors-on-libraries-that-depend-on-c-modules).

**Comment by drvinceknight** [comment]
> For future reference, I was only able to build my documentation by specifying an empty requirements file in RTD Admin, and by mocking any 3rd party packages in conf.py, as is described in the FAQ.

I tried this approach (and a large number of others) for #1671 but without any success.

**Comment by capcarr** [comment]
What settings are you using in RTD?

I have these checked, under the Advanced Settings:
* Install your project inside a virtualenv
* Give the virtual environment access to the global site-packages dir

And the requirements file is pointing to an empty file (actually two empty lines).

In the `conf.py` file I have (I'm using Python 2.7):

```
from mock import Mock as MagicMock

class Mock(MagicMock):
    @classmethod
    def __getattr__(cls, name):
            return Mock()

MOCK_MODULES = ['numpy', 'scipy', 'matplotlib', 'matplotlib.pyplot',
                'scipy.signal', 'scipy.interpolate', 'scipy.optimize',
                'scipy.stats', 'scipy.cluster', 'scipy.cluster.hierarchy',
                'scipy.cluster.vq', 'scipy.sparse', 'scipy.spatial',
                'scipy.spatial.distance', 'sklearn', 'sklearn.cluster',
                'sklearn.grid_search', 'sklearn.externals',
                'matplotlib.gridspec', 'h5py', 'shortuuid', 'bidict', 'svm',
                'sksvm']

sys.modules.update((mod_name, Mock()) for mod_name in MOCK_MODULES)
```

**Comment by drvinceknight** [comment]
I've tried that (I've been going through every single solution and permutation thereof I kind find). I've opened another issue #1671 in the hope of figuring this out (another core dev and I spent about 4 hours trying to fix this today).

Not sure if this is of interest but here's the build log: https://readthedocs.org/projects/axelrod/builds/3340162/

and here's a pic of my (latest) settings: https://www.dropbox.com/s/9ua3xq2ww8165y9/Screenshot%202015-09-21%2018.26.56.png?dl=0

for that my latest build failed with this mock in the conf (I copied exactly yours and added some matplotlib sub libraries, and then just had the ones I use: no difference all failed):

```
 18 from mock import Mock as MagicMock
 19
 20 class Mock(MagicMock):
 21     @classmethod
 22     def __getattr__(cls, name):
 23             return Mock()
 24
 25 MOCK_MODULES = ['numpy', 'scipy', 'matplotlib', 'matplotlib.pyplot',
 26                 'matplotlib.transforms', 'mpl_toolkits.axes_grid1',
 27                 'axelrod']
 28
 29 sys.modules.update((mod_name, Mock()) for mod_name in MOCK_MODULES)
```

I have also removed all uses of `autodoc` in case that was relevant. Thanks for answering: much appreciated.
